core/kernel.go

```diff
@@ -73,300 +65,305 @@ const (
+	kern.Service = rpc.NewService(accountState, nameRegState, kern.Blockchain, kern.State, nodeView, kern.Logger)
 
-				rpcevents.RegisterExecutionEventsServer(grpcServer, rpcevents.NewExecutionEventsServer(kern.State,
-					kern.Emitter, kern.Blockchain, kern.Logger))
+	if nodeView == nil {
+		checkTx := func(tx tmTypes.Tx, cb func(*abciTypes.Response)) error {
+			ctx := abci.TxExecutor("CheckTx", kern.exeChecker, kern.txCodec, kern.Logger.WithScope("CheckTx"))(tx)
+			dtx := abci.TxExecutor("DeliverTx", kern.exeCommitter, kern.txCodec, kern.Logger.WithScope("DeliverTx"))(tx)
+			appHash, err := kern.exeCommitter.Commit(nil)
+			if err != nil {
+				return err
+			}
+			kern.Blockchain.CommitBlock(time.Now(), nil, appHash)
+
+			cb(abciTypes.ToResponseCheckTx(ctx))
+			cb(abciTypes.ToResponseDeliverTx(abciTypes.ResponseDeliverTx{
+				Code:      dtx.Code,
+				Log:       dtx.Log,
+				Data:      dtx.Data,
+				Tags:      dtx.Tags,
+				GasUsed:   dtx.GasUsed,
+				GasWanted: dtx.GasWanted,
+				Info:      dtx.Info,
+			}))
+
+			cb(abciTypes.ToResponseCommit(abciTypes.ResponseCommit{
+				Data: appHash,
+			}))
+
+			return nil
+		}
 
-				rpcdump.RegisterDumpServer(grpcServer, rpcdump.NewDumpServer(kern.State,
-					kern.Blockchain, nodeView, kern.Logger))
+		kern.Transactor = execution.NewTransactor(kern.Blockchain, kern.Emitter,
+			execution.NewAccounts(kern.exeChecker, kern.keyClient, AccountsRingMutexCount),
+			checkTx, kern.txCodec, kern.Logger)
+	} else {
+		kern.Blockchain.SetBlockStore(bcm.NewBlockStore(nodeView.BlockStore()))
+		kern.Transactor = execution.NewTransactor(kern.Blockchain, kern.Emitter,
+			execution.NewAccounts(kern.exeChecker, kern.keyClient, AccountsRingMutexCount),
+			kern.Node.MempoolReactor().Mempool.CheckTx, kern.txCodec, kern.Logger)
+	}
+	return nil
+}
 
-				// Provides metadata about services registered
-				//reflection.Register(grpcServer)
+// KeysFromConfig sets the keyClient & keyStore based on the given config
+func (kern *Kernel) KeysFromConfig(conf *keys.KeysConfig) (err error) {
+	if conf.RemoteAddress != "" {
+		kern.keyClient, err = keys.NewRemoteKeyClient(conf.RemoteAddress, kern.Logger)
+		if err != nil {
+			return err
+		}
+	} else {
+		kern.keyStore = keys.NewKeyStore(conf.KeysDirectory, conf.AllowBadFilePermissions)
+		kern.keyClient = keys.NewLocalKeyClient(kern.keyStore, kern.Logger)
+	}
+	return nil
+}
 
-				go grpcServer.Serve(listen)
+// LoggerFromConfig adds a logging configuration to the kernel
+func (kern *Kernel) LoggerFromConfig(conf *logconfig.LoggingConfig) error {
+	logger, err := lifecycle.NewLoggerFromLoggingConfig(conf)
+	kern.SetLogger(logger)
+	return err
+}
 
-				return process.ShutdownFunc(func(ctx context.Context) error {
-					grpcServer.Stop()
-					// listener is closed for us
-					return nil
-				}), nil
-			},
-		},
+// ExecutorFromConfig builds the execution options for the kernel
+func (kern *Kernel) ExecutorFromConfig(conf *execution.ExecutionConfig) error {
```

Comment:
This should be renamed `LoadExecutionOptionsFromConfig`. It is not loading an exector